components/camel-zipkin/src/main/java/org/apache/camel/zipkin/ZipkinTracer.java

```diff
@@ -599,64 +622,55 @@ private void clientResponse(Brave brave, String serviceName, ExchangeSentEvent e
         }
     }
 
-    private ServerSpan serverRequest(Brave brave, String serviceName, Exchange exchange) {
-        ServerSpanThreadBinder serverBinder = brave.serverSpanThreadBinder();
-
+    private Span serverRequest(Tracing brave, String serviceName, Exchange exchange) {
         // reuse existing span if we do multiple requests from the same
         ZipkinState state = exchange.getProperty(ZipkinState.KEY, ZipkinState.class);
         if (state == null) {
             state = new ZipkinState();
             exchange.setProperty(ZipkinState.KEY, state);
         }
-        // if we started from a another server span then lets reuse that
-        ServerSpan last = state.peekServerSpan();
```

Comment:
ps if this behavior was needed (which would be odd to have a server child of a server!), you could do something like this:

```java
Span last = state.peekServerSpan();
Span span;
  if (last != null) {
    span = brave.tracer().newChild(last.context());
  } else {
    span = brave.tracer().nextSpan(EXTRACTOR.extract(exchange.getIn()));
  }
```